packages/cli/src/daemon/DaemonDecider.ts

```diff
@@ -118,6 +123,33 @@ export class DaemonDecider {
         }
     }
 
+    /**
+     * Record the process ID of the daemon that is being started for the current user.
+     * On a multi-user system, each user gets his/her own daemon.
+     *
+     * @private
+     * @memberof DaemonDecider
+     */
+    private recordDaemonPid() {
+        const pidForUser: IDaemonPidForUser = {
+            user: os.userInfo().username,
+            pid: process.pid
+        };
+
+        const pidFilePath = path.join(DaemonUtil.getDaemonDir(), "daemon_pid.json");
+        const pidForUserStr = JSON.stringify(pidForUser, null, 2);
+
+        try {
+            fs.writeFileSync(pidFilePath, pidForUserStr);
```

Comment:
Have we considered setting permission bits for the PID file? Should we use `644` to prevent other users from writing to the file, or is it safe to inherit the default permissions of the `~/.zowe/daemon` directory?

Reply:
We could force a 644 permission on the PID file. However, that would take some control out of the user's hands. The permissions of all new files is controlled by the UMASK setting. I am inclined to let the user determine how new files are created with the UMASK setting (via the umask command).

I could be convinced to explicitly set the file permissions ourselves if most people feel that is the better choice.